pkg/v1/remote/options.go

```diff
@@ -104,3 +106,14 @@ func WithPlatform(p v1.Platform) Option {
 		return nil
 	}
 }
+
+// WithCache is a functional option to specify a Cache implementation that
+// should be used when pulling remote image layers.
+//
+// It is only used for remote.Image.
+func WithCache(c cache.Cache) Option {
```

Comment:
This doesn't really enable anything that clients couldn't do by just wrapping the `remote.Image` themselves, right? I'm hesitant to add this as an option if we don't need it

Reply:
I thought it would be nice for usability, but then I tried it the other way and it was basically the same. Done.